### DISCUSSION TITLE
The first Codespaces check-in

### DISCUSSION THREAD

**Discussion by otter-computer** [community/community#33070] [Codespaces]
Hey everyone! I’m Thomas, your new Codespaces Community Manager. You may have seen me around the [Actions and Packages](https://github.com/orgs/community/discussions/categories/actions-and-packages) category, posting check-ins on Actions features and updates, but now I’m here to keep you in the loop on working on your code in the cloud!

For our first issue, I thought I’d bring you an update on all the new Codespaces features that have shipped over the past few months to help you and your team code quickly, directly from your browser. 

*Are you using Codespaces in your organisation? I’d love to hear about your setups in the comments below.*

Without further ado, let’s goooo~ 🚀

### [Codespace prebuilds](https://github.blog/2022-06-15-prebuilding-codespaces-is-generally-available/)
Prebuilds can rapidly speed up your Codespace creation, allowing you to start working on your code in record time–even with a massive repository! 

Prebuilds keep a “ready-to-go” template of your project where your source code, editor extensions, project dependencies, commands, and configurations have already been downloaded, installed, and applied–meaning you can spin up a Codespace and get to work in record time.

### [Organisation billing settings updates](https://github.blog/changelog/2022-08-23-organization-billing-settings-updates-for-codespaces/)
Organisation owners can now manage billing settings for org-owned repositories, allowing you to enable Codespaces for certain people in the organisation, everyone in the organisation, or all org members and outside collaborators. 

### [Retention policy](https://github.blog/changelog/2022-06-27-whats-new-in-codespaces-for-organizations/) and [default timeout](https://github.blog/changelog/2022-04-25-codespaces-now-offers-organizational-policies-to-restrict-idle-timeouts/) improvements
Organisation owners and individuals can now set a [default retention policy](https://github.blog/changelog/2022-06-27-whats-new-in-codespaces-for-organizations/) to manage how long unused Codespaces are retained allowing you to keep your cloud code environments squeaky clean!

Org owners can now also [set the default idle timeout length](https://github.blog/changelog/2022-04-25-codespaces-now-offers-organizational-policies-to-restrict-idle-timeouts/) stopping inactive Codespaces from eating up valuable billing time.
## GitHub Universe is on the horizon 🪐
[GitHub Universe](https://githubuniverse.com/) is rapidly approaching, and this year it’s back *in person* at the Yerba Buena Center for the Arts in San Francisco on November 9th and 10th. Check out the GitHub blog for a [sneak peek at what to expect](https://github.blog/2022-08-31-what-you-can-expect-at-github-universe-2022-cloud-security-community-and-ai/).

Not able to make it to the in-person event? No sweat! The daily keynotes, 80+ learning sessions, and hands-on content will be streamed online for free! [Register now](https://register.githubuniverse.com/) for access to everything Universe~ 🚀✨

—

That’s all for this time! Remember to let me know how your organization is using Codespaces in the comments down below and I’ll see you next time.

**Comment by DatDerpyDude81** [comment]
Hmm GitHub universe

**Comment by kidonng** [comment]
The retention period is very generous but I'd like to see some improvements:

- There's only one reminder email, sent 24 hours prior to a codespace being deleted, which is easy to miss.

  Ideally GitHub would also send another email earlier (say, 72 hours before retention ends), and also send a web notification.

- When deleting a codespace with uncommitted changes, it would be awesome to export the changes to a branch automatically.

  I just lost some work I simply forgot to finish, not too important but is going to take some time to redo.

  Since GitHub is acknowledging the unsaved changes in the reminder email and already provides the export to branch feature in the web UI, why not do it automatically?

  I think that's a reasonable default and I would happily accept if GitHub shortens the maximum retention period to a shorter period in the future (I hope not), knowing my work would not easily vanish.

Most of the work I mentioned can be automated in userspace, but official support would be much appreciated.
